### PR TITLE
Compat RF - Field Rations - Watersource Offsets

### PR DESCRIPTION
**Pull request by OverlordZorn**
Adds config properties for reaction forces vehicles which can carry drinking water.

### CODE REVIEW

**Comment by OverlordZorn** [comment]
For the Typhoon Water Carrier, i have an optional alternative which is a hatch on the right side, as this would make more sense instead of just in the rear, but:

- this hatch might be for the fuel tank of the vehicle.
- having the accesspoint rear-center might be more intuitive for a normal user.

File: addons/compat_rf/CfgVehicles.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class CfgVehicles {
+    // applies the offset to all RF Offroads which can have the optional Tank in the back
+    class Offroad_01_unarmed_base_F;
+    class Pickup_01_base_rf : Offroad_01_unarmed_base_F
+    {
+        acex_field_rations_offset[] = {-0.04, -2.45, -0.9};
```
Comment (rautamiekka) [inline addons/compat_rf/CfgVehicles.hpp:6]:
```suggestion
        XGVAR(offset)[] = {-0.04, -2.45, -0.9};
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    };
+    
+    // Enable Water Source by Default
+    class C_IDAP_Pickup_rf;
+    class C_IDAP_Pickup_water_rf: C_IDAP_Pickup_rf
+    {
+        acex_field_rations_waterSupply = 500;
```
Comment (rautamiekka) [inline addons/compat_rf/CfgVehicles.hpp:13]:
```suggestion
        XGVAR(waterSupply) = 500;
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    };
+    
+    class O_Truck_03_fuel_F;
+    class C_Truck_03_water_rf: O_Truck_03_fuel_F
+    {
+        acex_field_rations_waterSupply = 10000;
+        acex_field_rations_offset[] = {0, -5.05, -0.3}; // Rear
+        //acex_field_rations_offset[] = {1.25,-1.59,-0.58}; // Right Side Latch
```
Comment (rautamiekka) [inline addons/compat_rf/CfgVehicles.hpp:21]:
```suggestion
        XGVAR(waterSupply) = 10000;
        XGVAR(offset)[] = {0, -5.05, -0.3}; // Rear
        //XGVAR(offset)[] = {1.25,-1.59,-0.58}; // Right Side Latch
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    };
+
+    class B_Truck_01_fuel_F;
+    class C_Truck_01_water_rf : B_Truck_01_fuel_F 
+    {
+        acex_field_rations_waterSupply = 10000;
+        acex_field_rations_offset[] = {-0.41,-5.15,-0.3};
```
Comment (rautamiekka) [inline addons/compat_rf/CfgVehicles.hpp:28]:
```suggestion
        XGVAR(waterSupply) = 10000;
        XGVAR(offset)[] = {-0.41,-5.15,-0.3};
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]
Switching over to ACEX macros. I think I got it right.

**Comment by OverlordZorn** [comment]
> Switching over to ACEX macros. I think I got it right.
If this would be inside the ACEX Field_Rations Addon itself, it would work. but its inside the compat_rf addon.
The macro would resolve to `acex_compat_rf_offset`, so this would not work.

**Comment by OverlordZorn** [comment]
you would need to use  `EXGVAR(var1,var2)` as in `EXGVAR(field_ration,offset)`

File: addons/compat_rf/CfgVehicles.hpp [hunk 5]
```diff
@@ -0,0 +1,30 @@
+class CfgVehicles {
+    // applies the offset to all RF Offroads which can have the optional Tank in the back
+    class Offroad_01_unarmed_base_F;
+    class Pickup_01_base_rf : Offroad_01_unarmed_base_F
+    {
```
Comment (PabstMirror) [inline addons/compat_rf/CfgVehicles.hpp:5]:
```suggestion
    class Pickup_01_base_rf : Offroad_01_unarmed_base_F {
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        EXGVAR(field_rations,offset)[] = {-0.04, -2.45, -0.9};
+    };
+    
+    // Enable Water Source by Default
+    class C_IDAP_Pickup_rf;
+    class C_IDAP_Pickup_water_rf: C_IDAP_Pickup_rf
+    {
```
Comment (PabstMirror) [inline addons/compat_rf/CfgVehicles.hpp:12]:
```suggestion
    class C_IDAP_Pickup_water_rf: C_IDAP_Pickup_rf {
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        EXGVAR(field_rations,waterSupply) = 500;
+    };
+    
+    class O_Truck_03_fuel_F;
+    class C_Truck_03_water_rf: O_Truck_03_fuel_F
+    {
```
Comment (PabstMirror) [inline addons/compat_rf/CfgVehicles.hpp:18]:
```suggestion
    class C_Truck_03_water_rf: O_Truck_03_fuel_F {
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        EXGVAR(field_rations,waterSupply) = 10000;
+        EXGVAR(field_rations,offset)[] = {0, -5.05, -0.3}; // Rear
+        //EXGVAR(field_rations,offset)[] = {1.25,-1.59,-0.58}; // Right Side Latch
+    };
+
+    class B_Truck_01_fuel_F;
+    class C_Truck_01_water_rf : B_Truck_01_fuel_F 
+    {
```
Comment (PabstMirror) [inline addons/compat_rf/CfgVehicles.hpp:26]:
```suggestion
    class C_Truck_01_water_rf : B_Truck_01_fuel_F {
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    class C_IDAP_Pickup_rf;
+    class C_IDAP_Pickup_water_rf: C_IDAP_Pickup_rf {
+        EXGVAR(field_rations,waterSupply) = 500;
+    };
+    
+    class O_Truck_03_fuel_F;
+    class C_Truck_03_water_rf: O_Truck_03_fuel_F {
+        EXGVAR(field_rations,waterSupply) = 10000;
+        EXGVAR(field_rations,offset)[] = {0, -5.05, -0.3}; // Rear
+        //EXGVAR(field_rations,offset)[] = {1.25,-1.59,-0.58}; // Right Side Latch
```
Comment (Mike-MF) [inline addons/compat_rf/CfgVehicles.hpp:18]:
```suggestion
        EXGVAR(field_rations,offset)[] = {0, -5.05, -0.3};
```

The right side latch is more than likely the fuel tank.

File: addons/compat_rf/CfgVehicles.hpp [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    };
+
+    class B_Truck_01_fuel_F;
+    class C_Truck_01_water_rf : B_Truck_01_fuel_F {
+        EXGVAR(field_rations,waterSupply) = 10000;
+        EXGVAR(field_rations,offset)[] = {-0.41,-5.15,-0.3};
```
Comment (Mike-MF) [inline addons/compat_rf/CfgVehicles.hpp:24]:
```suggestion
        EXGVAR(field_rations,offset)[] = {-0.41, -5.15, -0.3};
```

**Review by Mike-MF — APPROVED** [review]

File: addons/compat_rf/CfgVehicles.hpp [hunk 11]
```diff
@@ -0,0 +1,25 @@
+class CfgVehicles {
+    // applies the offset to all RF Offroads which can have the optional Tank in the back
+    class Offroad_01_unarmed_base_F;
+    class Pickup_01_base_rf : Offroad_01_unarmed_base_F {
```
Comment (johnb432) [inline addons/compat_rf/CfgVehicles.hpp:4]:
```suggestion
    class Pickup_01_base_rf: Offroad_01_unarmed_base_F {
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    };
+    
+    class O_Truck_03_fuel_F;
+    class C_Truck_03_water_rf: O_Truck_03_fuel_F {
+        EXGVAR(field_rations,waterSupply) = 10000;
+        EXGVAR(field_rations,offset)[] = {0, -5.05, -0.3};
+    };
+
+    class B_Truck_01_fuel_F;
+    class C_Truck_01_water_rf : B_Truck_01_fuel_F {
```
Comment (johnb432) [inline addons/compat_rf/CfgVehicles.hpp:21]:
```suggestion
    class C_Truck_01_water_rf: B_Truck_01_fuel_F {
```

File: addons/compat_rf/CfgVehicles.hpp [hunk 13]
```diff
@@ -0,0 +1,25 @@
+class CfgVehicles {
+    // applies the offset to all RF Offroads which can have the optional Tank in the back
```
Comment (johnb432) [inline addons/compat_rf/CfgVehicles.hpp:2]:
```suggestion
    // Applies the offset to all RF Offroads which can have the optional tank in the back
```

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**